### DISCUSSION TITLE
Diagnostics/autocomplete toggle

### DISCUSSION THREAD

**Discussion by JeviScript** [NvChad/NvChad#2193] [Q&A] [answered]
Hey, neovim newbie here. I installed the NvChad and I am lost.

Is there a way to make it so lsp diagnostics messages on the ui would appear only when I press some shortcut, i.e. make it togglable. The same question for autocomplition. It's stlightly distracting for me when diagnostics or autocomplition pops up when I type. I want those to appear only when I ask so

**Comment by antoineco** [comment, marked as answer]
You can prevent autocompletion pop-ups from showing automatically by configuring the plugin as follows inside your `custom/plugins.lua` file:

```lua
local plugins = {
  {
    "hrsh7th/nvim-cmp",
    opts = {
      completion = {
        autocomplete = false,
      },
    },
  },
}

return plugins
```

After that, completion only triggers on `cmp.mapping.complete`, which is [mapped](https://github.com/NvChad/NvChad/blob/08f3deb9010c259dc3e6e060bbc49568cbcc50ba/lua/plugins/configs/cmp.lua#L76) to `<C-Space>` in NvChad.

More information at https://github.com/hrsh7th/nvim-cmp/blob/main/doc/cmp.txt

---

LSP diagnostic messages as virtual text can be disabled by configuring the appropriate LSP handler(s):

```lua
vim.lsp.handlers["textDocument/publishDiagnostics"] = vim.lsp.with(vim.lsp.diagnostic.on_publish_diagnostics, {
  virtual_text = false,
})
```

After that, diagnostic messages will only show on actions such as `vim.diagnostic.goto_next`, `vim.diagnostic.goto_prev` or `vim.diagnostic.open_float`, respectively [mapped](https://github.com/NvChad/NvChad/blob/v2.0/lua/core/mappings.lua) to `]d`, `[d` and `<leader>f` in NvChad.

Execute `:h lsp-handler-configuration` inside Neovim for more information about that topic.

**Reply by JeviScript** [reply]
Works like a charm, big thanks !

**Reply by brokenthorn** [reply]
This does not disable the popup like you say @antoineco.
For me it only disables autocomplete, which is exactly what I wanted in #2608.
Thanks!
